src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.cs

```diff
@@ -343,27 +343,27 @@ private async Task WriteEntryAsyncInternal(TarEntry entry, CancellationToken can
             switch (entry.Format)
             {
                 case TarEntryFormat.V7:
-                    entry._header._checksum = await entry._header.WriteAsV7Async(_archiveStream, buffer, cancellationToken).ConfigureAwait(false);
+                    await entry._header.WriteAsV7Async(_archiveStream, buffer, cancellationToken).ConfigureAwait(false);
```

Comment:
Now the checksum is written internally by the TarHeader methods, and don't have to resort to these workarounds.